Lib/shutil.py

```diff
@@ -1449,6 +1456,18 @@ def _access_check(fn, mode):
             and not os.path.isdir(fn))
 
 
+def _win32_need_current_directory_for_exe_path(cmd):
+    """
+    On Windows, we can use NeedCurrentDirectoryForExePathW to figure out
+    if we should add the cwd to PATH when searching for executables.
+
+    If we don't have ctypes, we'll fallback to old behavior which is to always add cwd.
+    """
+    if _CTYPES_SUPPORTED:
+        return bool(ctypes.windll.kernel32.NeedCurrentDirectoryForExePathW(cmd))
```

Comment:
I think implementing `_winapi.NeedCurrentDirectoryForExePath()` is necessary to guarantee documented behavior, whether or not ctypes is available. I can walk you through the steps for implementing that if you have some experience with C programming.

If ctypes does get used, use `ctypes.WinDLL('kernel32')` instead of `ctypes.windll.kernel32`. `ctypes.windll` can't be used since it caches `ctypes.WinDLL` instances, which cache function pointer objects. Some other package could set a custom `argtypes`, `restype`, or `errcheck` on the cached function pointer that breaks its use here.